numpy/core/numeric.py

```diff
@@ -2089,15 +2089,12 @@ def indices(dimensions, dtype=int):
     """
     dimensions = tuple(dimensions)
     N = len(dimensions)
-    if N == 0:
-        return array([], dtype=dtype)
+    shape = (1,)*N
     res = empty((N,)+dimensions, dtype=dtype)
     for i, dim in enumerate(dimensions):
```

Comment:
Good that you didn't do the `zip` -- it actually would break some corner cases (when an item is a scalar rather than an array).